asciidoc/modules/con_mg_improvements-in-vertx-http-client.adoc

```diff
@@ -0,0 +1,203 @@
+[id="improvements-in-vertx-http-client_{context}"]
+= Improvements in the {VertX} HTTP client
+
+This section describes the improvements in HTTP client.
+
+== HTTP client request and response methods take an asynchronous handler as input argument
+
+The `HttpClient` and `HttpClientRequest` methods have been updated to use asynchronous handlers. The methods take `Handler<AsyncResult<HttpClientResponse>>` as input instead of `Handler<HttpClientResponse>`.
+
+In earlier releases of {VertX}, the `HttpClient` methods `getNow()`, `optionsNow()` and `headNow()` used to return `HttpClientRequest`, that you had to further send to perform a request. The `getNow()`, `optionsNow()` and `headNow()` methods have been removed. In {VertX} {4}, you can directly send a request with the required information using `Handler<AsyncResult<HttpClientResponse>>`.
+
+The following examples show how to send a request in {VertX} {4}:
+
+* To perform a GET operation:
++
+[source,java,options="nowrap",subs="attributes+"]
+----
+Future<HttpClientResponse> f1 = client.get(8080, "localhost", "/uri", HttpHeaders.set("foo", "bar"));
+----
+
+* To POST with a buffer body:
++
+[source,java,options="nowrap",subs="attributes+"]
+----
+Future<HttpClientResponse> f2 = client.post(8080, "localhost", "/uri", HttpHeaders.set("foo", "bar"), Buffer.buffer("some-data"));
+----
+
+* To POST with a streaming body:
++
+[source,java,options="nowrap",subs="attributes+"]
+----
+Future<HttpClientResponse> f3 = client.post(8080, "localhost", "/uri", HttpHeaders.set("foo", "bar"), asyncFile);
+----
+
+However, you can use the `HttpClientRequest.request()` methods to create an `HttpClientRequest` instance that has not been sent to perform a request operation. These methods can be used in basic interactions such as:
+
+* Sending the request headers
+* HTTP/2 specific operations such as setting a push handler, setting stream priority, pings, and so on.
+* Creating a NetSocket tunnel
+* Providing fine grained write control
+* Resetting a stream
+* Handling 100 continue headers manually
+
+The following example shows you how to create an `HTTPClientRequest`.
+
+[source,java,options="nowrap",subs="attributes+"]
+----
+client.request(HttpMethod.GET, 8080, "example.com", "/resource", ar -> {
```

Comment:
@vietj Adam asked me add the method "HttpMethod.GET" in the first line of code:
client.request(HttpMethod.GET, 8080, "example.com", "/resource", ar -> {

Original note:
https://github.com/vert-x3/wiki/wiki/4.0.0-Deprecations-and-breaking-changes#%EF%B8%8F-request-based-api

Reply:
it is not clear what you mean, HttpMethod.GET is already present in the code

Reply:
@vietj I am sorry for the confusion. In the wiki page "HttpMethod.GET" is missing.
https://github.com/vert-x3/wiki/wiki/4.0.0-Deprecations-and-breaking-changes#%EF%B8%8F-request-based-api

Reply:
yes that's right indeed, I'll fix the wiki page

Reply:
@sangeetaraghu I updated the wiki page, thanks Adam